### DISCUSSION TITLE
More than 1 anat for 1 session with multiple runs

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#397] [General]
Hello,

How does rabies handle having more than 1 anat file and multiple func files within a session?

For example for 1 session, i.e. ses-01 we have the following:

1. 1st localizer, 1st t2 and 1st functional scan collected

Pull the animal out. Give an injection and then put animal back in the scanner.

2.2nd localizer, 2nd t2, and 2nd and 3rd functional scans collected.

Is there a way to setup RABIES and preprocess the data to where the 1st t2 and 1st functional are co-registered together and the 2nd t2 is co-registered together for ses-01?

Thanks.

-g

**Comment by Gab-D-G** [comment]
Hi,
If you set up the  --bids_filter option properly, you can select which BIDS spec will look for when selecting the anatomical. If you put only t2, then the t2 at each session will be selected as the anatomical basis for the pipeline, meaning it will coregister across session, and within session with the functional images.
Does that answer the question?

Gabriel

**Comment by geowk** [comment]
Gabriel,

Thanks for response and suggestion with the --bids_filter file. 

With the example online, would this work:

```
{
    "func": {
        "suffix":["bold","run-01"]
    },
    "anat": {
        "suffix":["T2w","run-01"]
    }
}

{
    "func": {
        "suffix":["bold","run-02","run-03","run-04"]
    },
    "anat": {
        "suffix":["T2w","run-02"]
    }
}
```

These are all from 1 session where:  T2 run-01 should be co-registered with func 01  and T2 run-02 should be co-registered with func 02, func 03, func 04.

-g

**Comment by Gab-D-G** [comment]
I understand better what you're trying to do now. It won't be possible to specify multiple runs of anatomical images within a single session. You will have to name them as different sessions. Perhaps you could call it ses-1a and ses-1b for example? You should not have to change the filter, because by default the T2w images are selected, which is what you want I believe.

**Comment by geowk** [comment]
Thanks, Gabriel. I broke them down into  ses-1a and ses-1b as you suggested and it worked that way. No additional changes needed.

Thanks, again.

-g
